src/librustc_mir/interpret/validity.rs

```diff
@@ -411,11 +411,11 @@ impl<'rt, 'mir, 'tcx, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, 'tcx, M
                     if !did.is_local() || self.ecx.tcx.is_foreign_item(did) {
                         return Ok(());
                     }
+                    // FIXME: Statics referenced from consts are skipped.
+                    // This avoids spurious "const accesses static" errors
+                    // unrelated to validity, but is similarly unsound.
                     if !self.may_ref_to_static && self.ecx.tcx.is_static(did) {
-                        throw_validation_failure!(
```

Comment:
Can't we keep this validation failure in case the static's memory is mutable? or do we already report errors if constants refer to mutable memory (i kinda remember talking about this with @RalfJung on a recent PR)

Reply:
The thing is, from a dynamic perspective there is nothing wrong with a const that *points to* a mutable static. So I don't think we should check for it here.

The static checks do that as part of their approximation, sure, but quoting from https://github.com/rust-lang/rust/issues/49146#issuecomment-616301045
> Miri is perfectly capable of emitting an error when it encounters an illegal operation on a qualified value, and it can do so with no false positives. 

I agree with @ecstatic-morse that it *should* do so with no false positives. But failing with a const *refers to* (but does not read from) a mutable static is a false positive.